### PR TITLE
feat: generate NFT metadata in memory

### PR DESCRIPTION
**Pull request by cazala**
The purpose of this PR is to stop hitting the marketplace subgraph to return metadata for parcels/estates, and instead generate all of the `NFT` objects while building the `Tile` objects on start up., and then serving them from memory.

In order to do so, instead of returning just `Tile[]`, the `api` component now returns `{ tiles: Tile[], parcels: NFT[], estates: NFT[] }`. Also the `map` component now has three records: `parcels`, `estates` and `tokens`, which give easy access to all the `NFT` by using the newly added methods `getParcel(x, y)`, `getEstate(estateId)` and `getToken(contractAddress, tokenId)`

### CODE REVIEW

**Comment by cazala** [comment]
rebased

File: src/modules/api/component.ts [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
+import {
+  isExpired,
+  getProximity,
+  graphql,
+  capitalize,
+  buildFromEstates,
+} from './utils'
+import { coordsToId, specialTiles } from '../map/utils'
 
 const tileFields = `{
```
Comment (nachomazzara) [inline src/modules/api/component.ts:25]: should this be `parcelFields` as we are calling `ParcelFragment` the type?

File: src/modules/api/component.ts [hunk 2]
```diff
@@ -182,86 +225,207 @@ export function createApiComponent(components: {
...
     )
-    const updatedTiles = parcels.map(fromTileFragment)
+
+    const updatedTiles = parcels.map(buildTile)
+    const updatedParcels = parcels.map(buildParcel)
+    const updatedEstates = parcels
+      .map(buildEstate)
+      .filter((estate) => estate !== null) as NFT[]
 
     // The following piece adds tiles from updated Estates. This is necessary only for an Estate that get listed or delisted on sale, since that doesn't chagne the lastUpdatedAt property of a Parcel.
```
Comment (nachomazzara) [inline src/modules/api/component.ts:235]: Typo `chagne`

File: src/modules/map/component.ts [hunk 3]
```diff
@@ -31,20 +36,64 @@ export function createMapComponent(components: {
     // mutations ahead (for performance reasons)
     for (const tile of newTiles.sort(sortByCoords)) {
       oldTiles[tile.id] = tile
-      lastUpdatedAt = Math.max(lastUpdatedAt, tile.updatedAt)
       computeEstate(tile.x, tile.y, oldTiles)
     }
     return oldTiles
   }
+  function addParcels(newParcels: NFT[], oldParcels: Record<string, NFT>) {
```
Comment (nachomazzara) [inline src/modules/map/component.ts:43]: should we need to add a blank here?

**Review by nachomazzara — CHANGES_REQUESTED** [review]
Excellent @cazala. Just minor comments

**Comment by cazala** [comment]
Thanks for the review @nachomazzara :) I addressed the feedback

**Pull request merged**